### DISCUSSION TITLE
Scaling Ethereum Hackathon

### DISCUSSION THREAD

**Discussion by gumdropsteve** [gumdropsteve/SmartRPA#31] [Ideas]
Reference: https://www.notion.so/Prizes-Scaling-Ethereum-1d3a6c2966dd4d8fa679d78890f2f8a6

# Initial thoughts
Seems obvi use,

#### Protocol Labs
> Protocol Labs is an open-source research, development, and deployment laboratory. Our projects include IPFS, Filecoin, libp2p, and many more. We aim to make human existence orders of magnitude better through technology.

#### Ideas
> - 🌸**NFT Metadata Updater** Mo minting, mo problems. Best method for managing signed permissioned and timestamped updates to NFT metadata stored on IPFS via any L2 chain.
  - this makes sense
> - 🏁**Filecoin Turbo Retrieval** Best method of bundling, managing, warming up, pre-flighting or otherwise speeding up payment channel process for easy mass retrieval from Filecoin.
  - this may make sense

## Chainlink - Price Feeds
> Chainlink is a decentralized network of oracles that enables smart contracts to securely interact with real-world data and services that exist outside of blockchain networks. Chainlink powers some of the top DeFi projects including Aave and Sythetix.

#### Ideas
> - Building a smart contract/decentralized application/blockchain based application using Chainlink oracles. To qualify, you need to make a state change on-chain based on something from Chainlink oracles.
  - #5 
> - Improving the Chainlink developer ecosystem. This is a tool that makes the lives of Chainlink smart contract engineers better.

## The Graph
yeah

# TBD...
Not sure.. caught attention,

## SKALE
> SKALE is an Ethereum native multi-blockchain network. SKALE is built to provide each Dapp with its own decentralized gasless blockchain of optimal size and capacity. Security and network orchestration mechanism are based directly on the Ethereum Mainnet. SKALE is full Solidity, very easy for developers to build on and for the end user to interact with.

- "Scalability and NFT" – Chadwick Strange, SKALE Labs
  - https://youtu.be/bxTXhRt0-Vg
  - Will a dapp run on different chains, or only on 1 chain? How is the interoperability of the chains? 
    - https://youtu.be/bxTXhRt0-Vg?t=1269

#### Ideas
> Uniswap on SKALE, NFT Platform on SKALE. Use your creativity, there are no restrictions on what to build, all ideas are welcome. Impress us ;)

### NuCypher 
> The NuCypher Network provides accessible, intuitive, and extensible runtimes and interfaces for secrets management, secure data exchange, dynamic access control, and secure computation.

![image](https://user-images.githubusercontent.com/43570913/115531861-18115200-a24a-11eb-87b4-aa1626e135e4.png)
https://youtu.be/MTyd3x7gp88?t=768

#### Ideas
> NuCypher will award a prize(s) for the most interesting project that leverages L2 solutions with NuCypher
> - Must use NuCypher technology in your submission
> - Must have an open-source license

### Secret Network
> [Secret Network](https://scrt.network/developers) is the first and only privacy preserving smart contracting platform. Secret Network introduces secret contracts, which are smart contracts with encrypted inputs, encrypted outputs, and encrypted state. Secret Contracts enable significant usability improvements (such as front-running resistance) and power a new breed of dApps (like access control, multiplayer gaming) that hasn't been possible before due to the public nature of blockchains.

#### Ideas
> When judging projects - we will be looking at the following aspects:
> - Secret Network mainnet traction potential (TVL, # of users, # of transactions)
> - Creativity
> - User experience
> - Novelty

Could be useful useful for transaction details

> We encourage teams to start projects that they'd be interested in continuing after the hackathon. Here's [list of projects](https://scrt.network/blog/announcing-secret-network-grant-program) we would like to see get build on Secret Network.

## Superfluid 
> Superfluid is a new DeFi Primitive that automates recurring transactions by allowing tokens to flow seamlessly from one wallet to another on Ethereum. It enables a variety of functions: from money streaming to one-transaction reward distributions, smart contract callbacks, a user level batch-call system, and an SDK for dApps. NFTs for land ownership in Decentraland? Automatically get rent paid in a stream! If you transfer the NFT the stream redirects with it NFTs for music rights? Get royalties paid directly to the NFT holder. NFT of your favourite art? Get social token rewards from the artist!

#### Ideas
> - Be a Superfluid magician and create a crazy cross-chain application. Example: open a stream on Polygon (Matic) by sending a transaction on another chain.
> - Explain it to Boomers! Build an amazing user experience, education tool or visualization tool to make streaming money easy to understand. Example: A visualization tool or a wacky game showing how money can move in real-time.
> - Build something awesome using Superfluid that makes money more fluid. Example: Create a tool for organizations to streamline discrete payments into constant flows.

**Comment by eserilev** [comment]
some ideas:

- Factionalized NFT ownership:
   - fungible tokens tied to a non fungible token
   - the fungible tokens represent fractionalized ownership  of the non fungible token
   - example: an NFT thats tied to 100 fungible tokens. each single fungible token represents a 1% ownership in the NFT
   - could introduce some Graph usecases?
   - https://github.com/ethereum/EIPs/issues/1634
   - https://github.com/tallyxinc/top-rft/blob/master/contracts/token/rft/RFT.sol

- Skale + IPFS idea: decentralized document storage and signature
   - NFT document contract (with possibly fractionalized document ownership capabilities?)
   - Initial NFT creation can be handled on eth mainnet
   - Updates to the NFT (i.e. document signatures) can be held on the Skale side chain to reduce fees?
   - maybe everything happens on Skale? im not sure what the implications of this would be
